src/ray/protobuf/instance_manager.proto

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.autoscaler;
+
+enum RayNodeKind {
+  HEAD = 0;
+  WORKER = 1;
+  UNKNOWN = 2;
+}
+
+// A node type that's available for the cluster.
+message InstanceType {
+  bytes type_name = 1;
+  RayNodeKind node_kind = 2;
+  map<string, double> resources = 3;
+}
+
+message GetAvailableInstanceTypesRequest {}
+
+message GetAvailableInstanceTypesResponse {
+  repeated InstanceType instance_types = 1;
+  repeated int64 available_instances = 2;
+}
+
+// Represents the state of a launched instance.
+// An instance is considered launched as long as
+// it has a unique instance_id associated with it.
+//
+// Note a launched instance may be DEAD. In this case,
+// the state will be garbage collected after some timeout
+// period (by default 30 minutes).
+message Instance {
+    enum InstanceState {
```

Comment:
for the state transitions - do we have some doc somewhere? Is this gonna be the same as existing autoscaler's instance behaviour?